lib/rules/function-linear-gradient-no-nonstandard-direction/index.js

```diff
@@ -19,14 +19,22 @@ const meta = {
 	url: 'https://stylelint.io/user-guide/rules/function-linear-gradient-no-nonstandard-direction',
 };
 
+const LINEAR_GRADIENT_FUNCTION = /(?:-webkit-|-moz-|-o-|-ms-)?linear-gradient\(/i;
+const LINEAR_GRADIENT_FUNCTION_NAME = /^(?:-webkit-|-moz-|-o-|-ms-)?linear-gradient$/i;
+
+const DIRECTION_WITH_TO = /^to (top|left|bottom|right)(?: (top|left|bottom|right))?$/i;
+const DIRECTION_WITHOUT_TO = /^(top|left|bottom|right)(?: (top|left|bottom|right))?$/i;
+
+const LEFT_RIGHT_TOP_BOTTOM = /left|right|top|bottom/i;
```

Comment:
reminder

Not worth it in this case, but you could reuse `LEFT_RIGHT_TOP_BOTTOM.source` in `DIRECTION_WITH_TO` and  `DIRECTION_WITHOUT_TO` using `RegExp` constructor. If you ever have a very long list, it would be cleaner/shorter.

Reply:
Agree. Considering maintainability (and if it never worsens performance), the following refactoring might be worth:

```js
const LINEAR_GRADIENT_FUNCTION = '(?:-webkit-|-moz-|-o-|-ms-)?linear-gradient';
const LINEAR_GRADIENT_FUNCTION_CALL = new RegExp(`${LINEAR_GRADIENT_FUNCTION}\\(`, 'i');
const LINEAR_GRADIENT_FUNCTION_NAME = new RegExp(`^${LINEAR_GRADIENT_FUNCTION}$`, 'i');

const DIRECTION = /top|left|bottom|right/;
const DIRECTION_WITH_TO = new RegExp(`^to (${DIRECTION.source})(?: (${DIRECTION.source}))?$`, 'i');
const DIRECTION_WITHOUT_TO = new RegExp(`^(${DIRECTION.source})(?: (${DIRECTION.source}))?$`, 'i');
```

Any thoughts?

Reply:
Thank you both for this suggestion 🙇